tests/subsys/settings/functional/nvs/CMakeLists.txt

```diff
@@ -0,0 +1,18 @@
+# SPDX-License-Identifier: Apache-2.0
+
+cmake_minimum_required(VERSION 3.13.1)
+include($ENV{ZEPHYR_BASE}/cmake/app/boilerplate.cmake NO_POLICY_SCOPE)
+project(NONE)
+
+FILE(GLOB app_sources ../src/*.c)
+target_sources(app PRIVATE ${app_sources})
```

Comment:
@kapi-no" the ".." parent directory above triggers CMake issue https://gitlab.kitware.com/cmake/cmake/issues/19475 and causes a reproducibility regression: it makes the build directory very long and depending on the source directory. See a similar example and more details (including ASCII art) at fix https://github.com/zephyrproject-rtos/hal_nordic/pull/6

Now unlike hal_nordic/pull/6, I bet there are other ways to fix this that do not necessarily require a new zephyr_library(). For instance these source files could simply be added from a higher level `CMakeLists.txt` which would be more consistent with normal CMake practices.

Can people familiar with this part of the code tell what is the proper way to organize it? Thanks.